script.js

```diff
@@ -1,4 +1,194 @@
+      outcome = tie;
+      games = games + 1;
+    } else if (
+      (userChoice == "scissors" && computerOutcome == "paper") ||
+      (userChoice == "paper" && computerOutcome == "stone") ||
+      (userChoice == "stone" && computerOutcome == "scissors")
+    ) {
+      outcome = lose;
+      loss = loss + 1;
+      games = games + 1;
+    } else {
+      outcome = invalid;
+    }
+  } else if (gameState == "Korean") {
+    // check user and computer inputs
+    if (input == "random") {
+      var userChoice = generateOutcome();
+    } else {
+      var userChoice = input;
+    }
+    var computerOutcome = generateOutcome();
+    console.log("Input: " + userChoice);
+    console.log("Generated: " + computerOutcome);
+
+    // the default outcome of all games shall be invalid
+    var outcome = invalid;
+
+    // If else logic for korean mode
+    if (
+      (userChoice == "scissors" && computerOutcome == "stone") ||
+      (userChoice == "paper" && computerOutcome == "scissors") ||
+      (userChoice == "stone" && computerOutcome == "paper")
+    ) {
+      previousWinner = "computer";
+      outcome =
+        "A draw in the next round would see the computer win this version of scissors, paper, stone.";
+    } else if (userChoice == computerOutcome) {
+      if (previousWinner == "computer") {
+        outcome = lose;
+        games = games + 1;
+        loss = loss + 1;
+        previousWinner = "No one";
+      } else if (previousWinner == playerName) {
+        outcome = win;
+        games = games + 1;
+        wins = wins + 1;
+        previousWinner = "No one";
+      } else {
+        outcome = tie;
+      }
+    } else if (
+      (userChoice == "scissors" && computerOutcome == "paper") ||
+      (userChoice == "paper" && computerOutcome == "stone") ||
+      (userChoice == "stone" && computerOutcome == "scissors")
+    ) {
+      previousWinner = playerName;
+      outcome =
+        "A draw in the next round allow you to win this version of scissors, paper, stone.";
+    } else {
+      outcome = invalid;
+    }
+    console.log("PreviousWinner:", previousWinner);
+    console.log("Player Name:" + playerName);
+    console.log("Boolean Check: " + previousWinner == playerName);
+  }
+
+  // Check current performance
+  var performance = wins / games;
+  if (performance >= 0.75) {
+    var evaluation = "Very impressive!";
+  } else if (performance <= 0.25) {
+    var evaluation = "Try harder!";
+  } else {
+    var evaluation = "Keep going!";
+  }
+
+  // Print output of game outcome
+  myOutputValue =
+    `You chose ${userChoice}. <br>The computer generated ${computerOutcome}<br><br>` +
+    outcome +
+    `<br><br>So far ${playerName}, your record is as follows. ${evaluation}<br>
+    Games: ${games}<br>
+    Wins: ${wins}<br>
+    Loss: ${loss}<br>
+    Draw: ${games - wins - loss}<br><br>
+    Now you can type "scissors", "paper", or "stone" to play another round!`;
   return myOutputValue;
```

Comment:
I liked that the code was readable due to proper variable namings as well as adequate comments. However, I think that it could be easier to read if you pull out your logic intensive game modes as separate helper functions that you call within the main functions. This is because when we write functions, a general rule of thumb will be to not write anything that spans more than 3/4 a screen. This is so that it is easier to understand for others and for yourself too when you go back to read  your code. By abstracting the logic into helper functions, you would no longer have to read through each line of specific logic to understand what the function does.